### PR TITLE
Switching to Managed Colored Console for mono support

### LINKED ISSUE #215: Mono and the Colored Console Appender
$ mono rh.exe -d=OldHotness -f=BOB --env=TEST --dryrun
log4net:ERROR Could not create Appender [ColoredConsoleAppender] of type [log4net.Appender.ColoredConsoleAppender]. Reported error follows.
System.EntryPointNotFoundException: GetConsoleOutputCP
  at (wrapper managed-to-native) log4net.Appender.ColoredConsoleAppender:GetConsoleOutputCP ()
  at log4net.Appender.ColoredConsoleAppender.ActivateOptions () [0x00000] in <filename unknown>:0
  at log4net.Repository.Hierarchy.XmlHierarchyConfigurator.ParseAppender (System.Xml.XmlElement appenderElement) [0x00000] in <filename unknown>:0
log4net:ERROR Appender named [ColoredConsoleAppender] not found.
log4net:ERROR Could not create Appender [ColoredConsoleAppender] of type [log4net.Appender.ColoredConsoleAppender]. Reported error follows.
System.EntryPointNotFoundException: GetConsoleOutputCP
  at (wrapper managed-to-native) log4net.Appender.ColoredConsoleAppender:GetConsoleOutputCP ()
  at log4net.Appender.ColoredConsoleAppender.ActivateOptions () [0x00000] in <filename unknown>:0
  at log4net.Repository.Hierarchy.XmlHierarchyConfigurator.ParseAppender (System.Xml.XmlElement appenderElement) [0x00000] in <filename unknown>:0
log4net:ERROR Appender named [ColoredConsoleAppender] not found.

### PR DESCRIPTION
**Pull request by drusellers**
Closes #215

taken from:
https://github.com/chocolatey/choco/blob/master/src/chocolatey/infrastructure/logging/log4net.mono.config.xml#L6

@ChrisMissal

### CODE REVIEW

File: product/roundhouse/infrastructure.app/logging/log4net.config.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ﻿<?xml version="1.0"?>
+<!-- For colors that can be used with ManagedColoredConsoleAppender, see
+     http://msdn.microsoft.com/en-us/library/system.consolecolor(v=vs.110).aspx 
+-->
 <log4net>
-  <appender name="ColoredConsoleAppender" type="log4net.Appender.ColoredConsoleAppender">
+  <appender name="ColoredConsoleAppender" type="log4net.Appender.ManagedColoredConsoleAppender">
     <mapping>
       <level value="ERROR" />
       <foreColor value="White, HighIntensity" />
```
Comment (ferventcoder) [inline product/roundhouse/infrastructure.app/logging/log4net.config.xml]: Note I'm not sure if high intensity is allowed. https://github.com/chocolatey/choco/blob/master/src/chocolatey/infrastructure/logging/log4net.mono.config.xml#L7-L34

Comment (drusellers) [inline product/roundhouse/infrastructure.app/logging/log4net.config.xml]: Its not. Fixing now and running on my mono project to verify. I also had to upgrade log4net.

**Comment by drusellers** [comment]
@ferventcoder on my mac `/usr/share` is set to root/wheel permissions. We may want to detect the mono runtime and change the value of `--outputpath` to be `~/.roundhouse` instead.

```
Running RoundhousE v0.8.6.0 against localhost - localhost.
Looking in ./db for scripts to run.
Please press enter when ready to kick...
Access to the path "/usr/share/ChuckNorris" is denied.
System.UnauthorizedAccessException: Access to the path "/usr/share/ChuckNorris" is denied.
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at System.IO.DirectoryInfo.Create () in <filename unknown>:line 0
   at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo:Create ()
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at System.IO.DirectoryInfo.Create () in <filename unknown>:line 0
   at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo:Create ()
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at System.IO.DirectoryInfo.Create () in <filename unknown>:line 0
   at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo:Create ()
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at System.IO.DirectoryInfo.Create () in <filename unknown>:line 0
   at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo:Create ()
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at System.IO.DirectoryInfo.Create () in <filename unknown>:line 0
   at (wrapper remoting-invoke-with-check) System.IO.DirectoryInfo:Create ()
   at System.IO.Directory.CreateDirectoriesInternal (System.String path) in <filename unknown>:line 0
   at System.IO.Directory.CreateDirectory (System.String path) in <filename unknown>:line 0
   at roundhouse.infrastructure.filesystem.WindowsFileSystemAccess.create_directory (System.String directory) in <filename unknown>:line 0
   at roundhouse.runners.RoundhouseMigrationRunner.create_change_drop_folder () in <filename unknown>:line 0
   at roundhouse.runners.RoundhouseMigrationRunner.run () in <filename unknown>:line 0
   at roundhouse.console.Program.run_migrator (ConfigurationPropertyHolder configuration) in <filename unknown>:line 0
   at roundhouse.console.Program.Main (System.String[] args) in <filename unknown>:line 0
```

File: product/roundhouse.tasks/app.config [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="NHibernate" publicKeyToken="aa95f207798dfdb4" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.3.1.4000" newVersion="3.3.1.4000" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (ferventcoder) [inline product/roundhouse.tasks/app.config]: Looks like an artifact. Please delete if possible.

File: product/roundhouse.databases.access/app.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<configuration>
+  <runtime>
+    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+      <dependentAssembly>
+        <assemblyIdentity name="NHibernate" publicKeyToken="aa95f207798dfdb4" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.3.1.4000" newVersion="3.3.1.4000" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (ferventcoder) [inline product/roundhouse.databases.access/app.config]: Please delete if possible.

File: product/roundhouse.tests.integration/app.config [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+        <assemblyIdentity name="Rhino.Mocks" publicKeyToken="0b3305902db7183f" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.6.0.0" newVersion="3.6.0.0" />
+      </dependentAssembly>
+      <dependentAssembly>
+        <assemblyIdentity name="NHibernate" publicKeyToken="aa95f207798dfdb4" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.3.1.4000" newVersion="3.3.1.4000" />
+      </dependentAssembly>
+    </assemblyBinding>
+  </runtime>
+</configuration>
```
Comment (ferventcoder) [inline product/roundhouse.tests.integration/app.config]: Please delete if possible. I'm less worried about this one.

File: product/roundhouse.tasks/app.config (hunk 2 above)
Comment (drusellers) [inline product/roundhouse.tasks/app.config]: Are you looking to fight the Nuget support? Every time you install a nuget its going to add this.

File: product/roundhouse.databases.access/app.config (hunk 3 above)
Comment (drusellers) [inline product/roundhouse.databases.access/app.config]: Are you looking to fight the Nuget support? Every time you install a nuget its going to add this.

File: product/roundhouse.tests.integration/app.config (hunk 4 above)
Comment (drusellers) [inline product/roundhouse.tests.integration/app.config]: Are you looking to fight the Nuget support? Every time you install a nuget its going to add this.

Comment (ferventcoder) [inline product/roundhouse.tests.integration/app.config]: Yeah, I don't have an issue deleting these. :)

File: product/roundhouse.tasks/app.config (hunk 2 above)
Comment (ferventcoder) [inline product/roundhouse.tasks/app.config]: This is one case where we don't want the app.config file.

**Comment by drusellers** [comment]
Ok, @ferventcoder please check again. the app.config changes have been removed.

**Comment by drusellers** [comment]
Closes #222

**Comment by ferventcoder** [comment]
I'm not sure this closes #222 (that is for user location logging when machine location is not available).

**Comment by drusellers** [comment]
Whoops. fixed.

**Comment by drusellers** [comment]
@ferventcoder @ChrisMissal similar to https://github.com/chucknorris/roundhouse/pull/186 - need for Mono support.

**Comment by ferventcoder** [comment]
Can we separate the upgrade to .net 4.5 from this PR please?

**Comment by BiggerNoise** [comment]
@drusellers I'm sorry to report that this P/R seems to have slid into merge oblivion. I have merge master into this branch locally. I can push that up if you want to start from there. However, looking over the changes, it seems like it might be faster to just create a new P/R from the current state of the code.

**Comment by BiggerNoise** [comment]
Nevermind @drusellers. It was a combination of a small issue and a invisibly locked DLL. Merge went in fairly cleanly after that.

**Comment by drusellers** [comment]
:100: YAY

**Pull request merged**